src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StageChunkFinder.java

```diff
@@ -22,7 +26,19 @@ public boolean isStartInsideChunk() {
 
     @Override
     public boolean isChunkStart(@Nonnull FlowNode current, @CheckForNull FlowNode previous) {
-        return current.getAction(LabelAction.class) != null && current.getAction(ThreadNameAction.class) == null;
+        if ((current instanceof StepAtomNode || current instanceof StepStartNode) && !((((StepNode) current).getDescriptor()) instanceof StageStep.DescriptorImpl)) {
```

Comment:
@jglick There's a LabelChunkFinder in the workflow-api PR that covers the all-labels case. 

If you do not filter down to stages then checkpoints will not work correctly, unfortunately.